hal/ia32/hal.c

```diff
@@ -58,6 +58,8 @@ void hal_init(void)
 	hal_consoleInit();
 	hal_exceptionsInit();
 	hal_interruptsInit();
-	hal_memoryInit();
+	if (hal_memoryInit() < 0) {
+		for (;;) { }
```

Comment:
Let's not heat the processor to the max, let it rest.
```suggestion
		for (;;) { 
			hal_cpuHalt();
		}
```